src/convertToObject.js

```diff
@@ -6,7 +6,25 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const resultObject = {};
+  let key = '';
+  let value = '';
+
+  const splitAllstring = sourceString.split(';');
+
+  for (const block of splitAllstring) {
+    const splitBlock = block.split(':');
+
+    key = splitBlock[0];
+    value = splitBlock[1];
+
+    if (value === undefined) {
+      continue;
+    }
+    resultObject[key.trim()] = value.trim();
```

Comment:
Avoid using loops like 'for...of' for this task. Instead, use array iteration methods such as 'map' or 'filter' to transform the array elements. This will help to make the code more functional in style and can improve readability.